src/main/java/core/basesyntax/AbstractFigure.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+public abstract class AbstractFigure implements Drawable, AreaCalculator {
+    private String type;

```

Comment:
I don't think we need this field actually, enum is used only for random figure generation pretty much, and classes implementations of the `AbstractFigure` are defined by the class name itself, and it is string, which is not ok if we have an enum